analytics/terraform/spark-k8s-operator/raydata-pipeline/main.tf

```diff
@@ -0,0 +1,313 @@
+      initial_workers = "2"
+    },
+    var.ray_config
+  )
+}
+
+# =============================================================================
+# Data Sources
+# =============================================================================
+
+# Get current AWS account ID
+data "aws_caller_identity" "current" {}
+
+# Get EKS cluster information
+data "aws_eks_cluster" "cluster" {
+  name = var.eks_cluster_name
+}
+
+# Extract OIDC issuer ID
+locals {
+  oidc_issuer_url = data.aws_eks_cluster.cluster.identity[0].oidc[0].issuer
+  oidc_issuer_id  = split("/", local.oidc_issuer_url)[4]
+  account_id      = data.aws_caller_identity.current.account_id
+}
+
+# =============================================================================
+# IAM Policy for S3 and Glue Access (for Iceberg)
+# =============================================================================
+
+data "aws_iam_policy_document" "s3_access" {
+  # Allow reading Spark logs from S3
+  statement {
+    effect = "Allow"
+    actions = [
+      "s3:GetObject",
+      "s3:GetObjectVersion"
+    ]
+    resources = [
+      "arn:aws:s3:::${var.s3_bucket}/${var.s3_prefix}*"
+    ]
+  }
+
+  # Allow writing Iceberg data to S3
+  statement {
+    effect = "Allow"
+    actions = [
+      "s3:GetObject",
+      "s3:PutObject",
+      "s3:DeleteObject",
+      "s3:ListMultipartUploadParts",
+      "s3:AbortMultipartUpload"
+    ]
+    resources = [
+      "arn:aws:s3:::${var.s3_bucket}/iceberg-warehouse/*"
+    ]
+  }
+
+  # Allow listing bucket with prefix restriction
+  statement {
+    effect = "Allow"
+    actions = [
+      "s3:ListBucket",
+      "s3:GetBucketLocation"
+    ]
+    resources = [
+      "arn:aws:s3:::${var.s3_bucket}"
+    ]
+    condition {
+      test     = "StringLike"
+      variable = "s3:prefix"
+      values = [
+        "${var.s3_prefix}*",
+        "iceberg-warehouse/*",
+        trimsuffix(var.s3_prefix, "/")
+      ]
+    }
+  }
+
+  # Additional ListBucket permission for exact prefix match
+  statement {
+    effect = "Allow"
+    actions = [
+      "s3:ListBucket"
+    ]
+    resources = [
+      "arn:aws:s3:::${var.s3_bucket}"
+    ]
+    condition {
+      test     = "StringEquals"
+      variable = "s3:prefix"
+      values = [
+        trimsuffix(var.s3_prefix, "/")
+      ]
+    }
+  }
+
+  # AWS Glue permissions for Iceberg catalog
+  statement {
+    effect = "Allow"
+    actions = [
+      "glue:GetDatabase",
+      "glue:GetDatabases",
+      "glue:CreateDatabase",
+      "glue:GetTable",
+      "glue:GetTables",
+      "glue:CreateTable",
+      "glue:UpdateTable",
+      "glue:DeleteTable",
+      "glue:GetPartition",
+      "glue:GetPartitions",
+      "glue:CreatePartition",
+      "glue:UpdatePartition",
+      "glue:DeletePartition",
+      "glue:BatchCreatePartition",
+      "glue:BatchDeletePartition",
+      "glue:BatchUpdatePartition"
+    ]
+    resources = [
+      "arn:aws:glue:${var.aws_region}:${local.account_id}:catalog",
```

Comment:
refer to the examples like this to remove var. etc.  https://github.com/awslabs/data-on-eks/blob/d59043897e922a854c64fa725eb6d822fc62ba70/analytics/terraform/spark-k8s-operator/main.tf#L163